### PR TITLE
#2170 Make sure that an import is created for constructor mapping defined variables

### LINKED ISSUE #2170: Missing import in generated mapping class when using generic mapper interface
When using generic mapper interface (eg: `EntityMapper` in below) is causing missing import.

### Entity & DTO
```java
@Setter
@Getter
@AllArgsConstructor
class Address{ 
    private String zipCode;
    List<Person> person;
}

@Setter
@Getter
@AllArgsConstructor
class Person { 
    String name;
}

@Data
class AddressDto {
    private String zipCode;
    List<PersonDto> person;
}

@Data
class PersonDto {
   String name;
}

```
### Mappers
```java
public interface EntityMapper<D, E> {
    E toEntity(D dto);
    D toDto(E entity);
    List<E> toEntity(List<D> dtoList);
    List<D> toDto(List<E> entityList);
}

@Mapper(uses = {PersonMapper.class})
public interface AddressMapper extends EntityMapper<AddressDto, Address> {
}

@Mapper
public interface PersonMapper extends EntityMapper<PersonDto, Person> {
}

```
### Result
Generated implementation mapper class `AddressMapperImpl` has no import of `Person` class.

### PR DESCRIPTION
**Pull request by filiphr**
Fixes #2170

### CODE REVIEW

File: processor/src/test/java/org/mapstruct/ap/test/bugs/_2170/mapper/AddressMapper.java [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+
+import org.mapstruct.Mapper;
+import org.mapstruct.ap.test.bugs._2170.dto.AddressDto;
+import org.mapstruct.ap.test.bugs._2170.entity.Address;
+import org.mapstruct.factory.Mappers;
+
+/**
+ * @author Filip Hrisafov
+ */
+//CHECKSTYLE:OFF
```
Comment (sjaakd) [inline processor/src/test/java/org/mapstruct/ap/test/bugs/_2170/mapper/AddressMapper.java:16]: Why is this needed?

Comment (filiphr) [reply processor/src/test/java/org/mapstruct/ap/test/bugs/_2170/mapper/AddressMapper.java:16]: Checkstyle was complaining about an interface with no methods

**Comment by filiphr** [comment]
Thanks for the review @sjaakd
